internal/service/flexcluster/resource.go

```diff
@@ -196,3 +205,27 @@ func splitFlexClusterImportID(id string) (projectID, clusterName *string, err er
 
 	return
 }
+
+func isUpdateAllowed(ctx context.Context, plan, state *TFModel) (bool, error) {
+	if !plan.ProjectId.Equal(state.ProjectId) {
+		return false, errors.New("project_id cannot be updated")
+	}
+	if !plan.Name.Equal(state.Name) {
+		return false, errors.New("name cannot be updated")
+	}
```

Comment:
Given we are opting for throwing a validation over using RequiresReplace() plan modifier, could be nice to implement a plan modifier that returns a validation when there are changes. This way we can use these across all resource implementation and not have custom code.

Reply:
I have done this, let me know what you think